### PR TITLE
Find list userguide

### PR DESCRIPTION
**Pull request by H1410101**
Updated user guide for the new find list functionality, and included word boundary matching functionality (previously forgot to PR).
I feel a bit lost when it comes to documentation, so please advise. Thanks!

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -123,11 +124,17 @@ Alias: `find`
 * Keywords can overlap. e.g. `samm my` will match `Sammy`
 * Fosters must match all keywords (i.e. `AND` search).
```
Comment (nananakx-x) [inline docs/UserGuide.md]: Fosterers...?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -123,11 +124,17 @@ Alias: `find`
...
   e.g. `Hans Bo` will return `Hansbo Grahm`, but not `Hans Duo`
+* Symbols between keywords or sections will combine them according to the function of the symbol.
+  `|` is `or`, `&` is `and` (low precedence), and ` ` (space) is `and` (high precedence).
+  e.g. `a & b | c d` is the same as `a & (b | (c & d))`
```
Comment (nananakx-x) [inline docs/UserGuide.md:170]: very clear example

File: docs/UserGuide.md [hunk 3]
```diff
@@ -123,11 +124,17 @@ Alias: `find`
...
+  e.g. `"Tom"` matches "Tom", but not "Tommy"
 
 Examples:
 * `list` lists all fosterers in the address book
 * `list john doe` matches "John Doe", "Doe John", "Johnny Doe", and "Mary" who lives on "John Doe Street"
-* `list john john doe` is redundant and gives the same result as `list john doe`
+* `find john john doe` is redundant and gives the same result as `find john doe`
+* `list "John" | zam & doe` matches "John Doe" and "Doe Shazam", but not "John Grahm"
 
 Expected output (success):
```
Comment (nananakx-x) [inline docs/UserGuide.md:180]: I think this is not the current success output if im not wrong, but if you are planning to change it to this then can just leave it just such! same for failure expected output

File: src/main/java/seedu/address/logic/search/FindCommandArgumentParser.java [hunk 4]
```diff
@@ -205,17 +205,40 @@ void findAndAppendNextPredicate() throws ParserDualStack.UnexpectedTokenExceptio
...
+                incrementCharIndex();
+            }
+            dualStack.append(SingleTextSearchMatcher.getQuotedMatch(
+                    search.substring(predicateStartIndex, index)
+            ));
+            if (hasChar() && isCharQuote()) {
+                incrementCharIndex();
+            }
+        }
+
```
Comment (nananakx-x) [inline src/main/java/seedu/address/logic/search/FindCommandArgumentParser.java:231]: lgtm

File: docs/UserGuide.md (hunk 1 above)
Comment (H1410101) [reply docs/UserGuide.md]: mb, added to next commit

File: docs/UserGuide.md (hunk 3 above)
Comment (H1410101) [reply docs/UserGuide.md:180]: I changed the doc to reflect the actual output, thanks!

File: docs/UserGuide.md [hunk 5]
```diff
@@ -155,25 +155,34 @@ Expected output (fail):
...
 * The keywords are case-insensitive.
 * The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`.
 * All fields are searched (including tags).
 * Keywords can match as parts of words. e.g. `john` will match `Johnny`.
 * Keywords can overlap. e.g. `samm my` will match `Sammy`
-* Fosters must match all keywords (i.e. `AND` search).
+* Fosterers must match all keywords (i.e. `AND` search).
   e.g. `Hans Bo` will return `Hansbo Grahm`, but not `Hans Duo`
+* Symbols between keywords or sections will combine them according to the function of the symbol.
+  `|` is `or`, `&` is `and` (low precedence), and ` ` (space) is `and` (high precedence).
```
Comment (nananakx-x) [inline docs/UserGuide.md:169]: do you want to change this to '/' now?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -155,25 +155,34 @@ Expected output (fail):
...
+  e.g. `a & b | c d` is the same as `a & (b | (c & d))`
+* Use double quotes `"` for exact, case-sensitive, word-level match. 
+  e.g. `"Tom"` matches "Tom", but not "Tommy"
 
 Examples:
 * `list` lists all fosterers in the address book
 * `list john doe` matches "John Doe", "Doe John", "Johnny Doe", and "Mary" who lives on "John Doe Street"
-* `list john john doe` is redundant and gives the same result as `list john doe`
+* `find john john doe` is redundant and gives the same result as `find john doe`
+* `list "John" | zam & doe` matches "John Doe" and "Doe Shazam", but not "John Grahm"
```
Comment (nananakx-x) [inline docs/UserGuide.md:178]: same for this, if you have already ensured its working as intended

**Review by nananakx-x — APPROVED** [review]
lgtm!

**Pull request merged**